client/sources/ok_test/sqlite.py

```diff
@@ -21,9 +23,46 @@ def load(self, code, setup='', teardown=''):
 
         Loads the sqlite3 module before loading any code.
         """
-        self.sqlite3 = self._import_sqlite()
         super().load(code, setup, teardown)
-        self._conn = self.sqlite3.connect(':memory:', check_same_thread=False)
+
+    def interpret(self):
+        """Interprets the code in this Console.
+
+        Due to inconsistencies with sqlite3 and Python's bindings, the following
+        process is used to interpret code:
+
+        1. If the Python sqlite3 module has an up-to-date binding (compared to
+           self.VERSION), interpret line-by-line with full output validation.
+        2. Otherwise, if there is an executable called "sqlite3" (in the current
+           directory is okay), pipe the test case into sqlite3 and display
+           expected and actual output. No output validation is available;
+           students have to verify their solutions manually.
+        3. Otherwise, report an error.
+        """
+        if self._import_sqlite():
+            self._conn = self.sqlite3.connect(':memory:', check_same_thread=False)
+            return super().interpret()
+        env = dict(os.environ,
+                   PATH=os.getcwd() + os.pathsep + os.environ["PATH"])
+        if self._has_sqlite_cli(env):
+            print('Unfortunately, OK is unable to use sqlite3 to test your code directly.')
+            print('Here is a transcript of what your code does in the sqlite3 interpreter.')
+            print()
+            test, expected, result = self._use_sqlite_cli(env)
+            print('TEST:')
+            print(format.indent(test, '    '))
+            print('EXPECTED (order does not matter):')
+            print(format.indent(expected, '    '))
+            print('OUTPUT:')
+            print(format.indent(result, '    '))
+            print()
+            print("Please manually check if your solution's output is correct.")
+            return False
+        else:
+            print("ERROR: could not run sqlite3.")
+            print("Tests will not pass, but you can still submit your assignment.")
+            print("Please download the newest version of sqlite3 to run tests.")
```

Comment:
I think it's good to mention downloading into the current folder.

However, I don't think it's good for ok to download the binary because we don't know which binary to download. We could do something fancy with the OS version and stuff, but it feels like it'd be too janky to actually work.